main.go

```diff
@@ -67,6 +61,23 @@ func checkError(e error) {
 	}
 }
 
+// Load the beacon log from file
+func loadLog() BeaconLog {
+	beaconLogData := new(BeaconLog)
```

Comment:
I think you are correct here. I will push some further changes which remove the explicit creation of a pointer.